src/core/validations/message.rs

```diff
@@ -262,6 +264,41 @@ fn validate_longitude(value: &str) -> Result<(), ValidationError> {
     Ok(())
 }
 
+pub fn validate_user_data_primary_address_ethereum(input: &String) -> Result<(), ValidationError> {
+    // Empty string is allowed for unsetting the primary address
+    if input.is_empty() {
+        return Ok(());
+    }
+
+    if !input.starts_with("0x") || input.len() != 42 {
+        return Err(ValidationError::InvalidDataLength);
+    }
+
+    let parsed = Address::from_hex(input).map_err(|_| ValidationError::InvalidData)?;
+
+    let checksummed = parsed.to_checksum(None);
```

Comment:
I noticed there are some better helper functions to validate Sol/Eth addresses. I'll swap my code out with those, where appropriate